### PR TITLE
Documentation for ft:highlight-field-matches

### PR DESCRIPTION
**Pull request by tuurma**
accompanying https://github.com/eXist-db/exist/pull/3423

### CODE REVIEW

**Comment by tuurma** [comment]
@line-o please note that util:expand and ft:highlight-field-matches are not directly comparable as they operate on different structures, so I don't see where would such a mention really belong. To more elaborate discussion on different ways of highlighting matches in kwic module perhaps, but not directly related to the PR in question.

File: src/main/xar-resources/data/lucene/lucene.xml [hunk 1]
```diff
@@ -590,6 +590,24 @@
...
+            <para>When retrieving the content of a field for display, you may still need an indication
+                 of where matches were found.</para>
+            <para>
+                <literal>ft:highlight-field-matches</literal> will return a copy of the XML fragment it
+                 received in its first parameter with <literal>exist:match</literal>
+                tag wrapped around the fields that match. Effectively it provides the same mechanism as
+                <literal>util:expand</literal> for full text matches but works on fields instead.</para>
+
+            <para>For example, to display each matching article with its title, author and matched full
+                text as KWIC, you could start with the code below. Please note that
```
Comment (line-o) [inline src/main/xar-resources/data/lucene/lucene.xml:604]: I would mention here, that kwic output is different from what you would get with `util:expand`.

**Comment by line-o** [comment]
@tuurma what you just wrote would be valuable info, as I was not aware that the structure is a completely different one.

**Comment by adamretter** [comment]
@tuurma I think @joewiz noted in the PR for the feature that the output is different and somewhat unexpected compare to util:expand. So that would definitely benefit from being documented - as he is only the first person to see the difference... others will struggle or see it too

**Comment by tuurma** [comment]
@line-o and @adamretter I do not agree, you can't use util:expand on a field, that's the whole point of the PR. @joewiz tried to compare but obviously he must have prepared his test so that the value used for the field would be stored in an XML node (in order to be able to use util:expand).

Therefore, nobody could possibly move from using util:expand to ft:highligh-field-matches and observe differences when passing the results to kwic. Adding a note like Juri suggested would only confuse matters and indicate that somehow util:expand and ft:highlight-field-matches are interchangeable when they definitely are not.

**Comment by adamretter** [comment]
> I do not agree, you can't use util:expand on a field, that's the whole point of the PR.

@tuurma Neither of us are saying that you can! We also understand the point of the PR!

**Comment by adamretter** [comment]
I think the suggestion that myself, @line-o and @joewiz are trying to make is that the expanded output from `util:expand` and `ft:highlight-field-matches` are similar, as you yourself noted in your docs update. The two functions are designed to do similar tasks with different inputs... yet the results are not the same. The difference in results is generally unexpected! 

We are only suggesting that it should be documented that it is different and where it differs... so that the "differences" don't surprise other users too!

**Comment by wolfgangmm** [comment]
Maybe add an explanation like the following:

* `util:expand` operates on nodes which are part of an XML document
* fields are attached to a node, but can contain arbitrary content which may be derived from the document, but does not need to

Thus `util:expand` and `ft:highlight-field-matches` behave differently and produce slightly different output. `ft:highlight-field-matches` is also faster as it does not need to traverse the XML tree like `util:expand`.

File: src/main/xar-resources/data/kwic/kwic.xml [hunk 2]
```diff
@@ -165,5 +165,8 @@
         <para> <literal>util:expand</literal> returns a copy of the XML fragment it received in its
             first parameter, which, unless configured otherwise, has all matches wrapped into
             <tag>exist:match</tag> tags.</para>
+
+        <para>Please note that <literal>util:expand</literal> will not expand matches in Lucene fields. 
+        Use <literal>ft:highlight-field-matches</literal> instead.</para>
```
Comment (joewiz) [inline src/main/xar-resources/data/kwic/kwic.xml:170]: One additional sentence pointing to the great description of the differences between the two functions would be helpful to add here: 

> For more information, see `lucene.xml#expand-fields`

Comment (tuurma) [reply src/main/xar-resources/data/kwic/kwic.xml]: Thanks, added now.

**Pull request merged**